src/Command/InfectionCommand.php

```diff
@@ -368,58 +371,65 @@ private function checkMetrics(): bool
+
+        $minMsi = $input->getOption('min-msi');
+
+        if (null !== $minMsi && $minMsi !== (string) (float) $minMsi) {
+            throw new InvalidArgumentException(sprintf('Expected min-msi to be a float. Got "%s"', $minMsi));
         }
 
-        /** @var string|null $mutators */
-        $mutators = $input->hasOption('mutators')
-            ? trim((string) $input->getOption('mutators'))
-            : null
-        ;
+        $minCoveredMsi = $input->getOption('min-covered-msi');
 
-        if ($mutators === '') {
-            $mutators = null;
+        if (null !== $minCoveredMsi && $minCoveredMsi !== (string) (float) $minCoveredMsi) {
+            throw new InvalidArgumentException(sprintf('Expected min-covered-msi to be a float. Got "%s"', $minCoveredMsi));
         }
 
+        $mutators = trim((string) $input->getOption('mutators'));
+
         $this->container = $this->getApplication()->getContainer()->withDynamicParameters(
-            $configFile,
-            $mutators,
-            (bool) $input->getOption('show-mutations'),
+            '' === $configFile ? null : $configFile,
+            '' === $mutators ? null : $mutators,
+            $input->getOption('show-mutations'),
             trim((string) $input->getOption('log-verbosity')),
-            (bool) $input->getOption('debug'),
-            (bool) $input->getOption('only-covered'),
-            (string) $input->getOption('formatter'),
-            (bool) $input->getOption('no-progress'),
-            $input->hasOption('coverage')
-                ? trim((string) $input->getOption('coverage'))
-                : '',
-            trim((string) $input->getOption('initial-tests-php-options') ?: ''),
-            (bool) $input->getOption('ignore-msi-with-no-mutations'),
-            (float) $input->getOption('min-msi'),
-            (float) $input->getOption('min-covered-msi')
+            $input->getOption('debug'),
+            $input->getOption('only-covered'),
+            trim((string) $input->getOption('formatter')),
+            $input->getOption('no-progress'),
+            '' === $coverage ? null : $coverage,
+            '' === $initialTestsPhpOptions ? null : $initialTestsPhpOptions,
+            $input->getOption('ignore-msi-with-no-mutations'),
+            null === $minMsi ? null : (float) $minMsi,
+            null === $minCoveredMsi ? null : (float) $minCoveredMsi,
+            '' === $testFramework ? null : $testFramework,
+            '' === $testFrameworkOptions ? null : $testFrameworkOptions
         );
     }
 
-    private function includeUserBootstrap(InfectionConfig $config): void
+    private function includeUserBootstrap(Configuration $config): void
     {
         $bootstrap = $config->getBootstrap();
 
-        if ($bootstrap) {
-            if (!file_exists($bootstrap)) {
-                throw FileNotFound::fromFileName($bootstrap, [__DIR__]);
-            }
+        if (null === $bootstrap) {
```

Comment:
The file existence is asserted upstream: IMO it should be done at the configuration level as part of the configuration validation rather than when using its values

Reply:
Actually kept the check so if to be changed, it should be done in a separate PR